src/nbody_base.jl

```diff
@@ -0,0 +1,157 @@
+using StaticArrays, DiffEqBase
+#Structures for bodies and systems under symulations
+
+# Fields for position, velocity and mass of a particle should be required for every descendant
```

Comment:
This might be a good use of https://github.com/tbreloff/ConcreteAbstractions.jl . We can fork and register if it makes sense. Or `@def`

Reply:
That is actually cool! Good, I'll check what I can do with @def first.